src/spark-core.js

```diff
@@ -143,7 +143,11 @@ var SparkCore = AmpersandState.extend({
       .catch(function logDeviceRemovalFailure(reason) {
         this.logger.warn(reason);
       })
-      .then(this.credentials.logout.bind(this.credentials));
+      .then(function cleanUpAndNotify() {
+        this.credentials.logout();
```

Comment:
Ah, gotcha, that could be. looking a little deeper, `Credentials#logout()` redirects to the CI logout page, so all other logout work needs to come before that.